### PR TITLE
Python: Let produce last-updated-ms field in metadata at microsecond level

### LINKED ISSUE #8025: Python: pyiceberg table metadata produce improper last-updated-ms value
### Apache Iceberg version

1.3.0 (latest release)

### Query engine

None

### Please describe the bug 🐞

pyiceberg metadata produces `last-updated-ms` in microsecond resolution, not millisecond.

This behavior violate iceberg table spec

### PR DESCRIPTION
**Pull request by ktk1012**
close #8025 

Use millisecond in TableMetadata in last-update-ms in construction.

### CODE REVIEW

File: python/pyiceberg/utils/datetime.py [hunk 1]
```diff
@@ -91,6 +91,25 @@ def timestamp_to_micros(timestamp_str: str) -> int:
...
+def datetime_to_millis(dt: datetime) -> int:
+    """Converts a datetime to milliseconds from 1970-01-01T00:00:00.000000."""
+    if dt.tzinfo:
+        delta = dt - EPOCH_TIMESTAMPTZ
+    else:
+        delta = dt - EPOCH_TIMESTAMP
+    return (delta.days * 86400 + delta.seconds) * 1_000 + delta.microseconds // 1_000
+
+
+def timestamp_to_millis(timestamp_str: str) -> int:
```
Comment (Fokko) [inline python/pyiceberg/utils/datetime.py:103]: Why add this one when we don't use it? I would rather leave it out.

File: python/pyiceberg/utils/datetime.py [hunk 2]
```diff
@@ -91,6 +91,25 @@ def timestamp_to_micros(timestamp_str: str) -> int:
     raise ValueError(f"Invalid timestamp without zone: {timestamp_str} (must be ISO-8601)")
 
 
+def datetime_to_millis(dt: datetime) -> int:
```
Comment (Fokko) [inline python/pyiceberg/utils/datetime.py:94]: Could you add an unit test for this one?

**Review by Fokko — COMMENTED** [review]
Thanks for fixing this! One comment, would it be possible to add a test? Apart from that it looks good to go!

File: python/pyiceberg/utils/datetime.py (hunk 1 above)
Comment (ktk1012) [reply python/pyiceberg/utils/datetime.py:103]: Okay, I'll leave out this

File: python/pyiceberg/utils/datetime.py (hunk 2 above)
Comment (ktk1012) [reply python/pyiceberg/utils/datetime.py:94]: OK. added some tests

File: python/tests/utils/test_datetime.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
+from datetime import datetime, tzinfo
```
Comment (Fokko) [inline python/tests/utils/test_datetime.py:1]: Can you add the Apache header here? Otherwise the tests will fail

Comment (ktk1012) [reply python/tests/utils/test_datetime.py:17]: Ah, okay. Added it

**Pull request merged**